src/components/messenger/list/utils/utils.test.ts

```diff
@@ -44,3 +44,20 @@ describe('isUserAdmin', () => {
     expect(isUserAdmin(user, adminIds)).toBe(false);
   });
 });
+
+describe('getUserHandle', () => {
+  it('returns primaryZID when it is present', () => {
+    const user = { primaryZID: 'zid123', wallets: [{ id: 'wallet-id-1', publicAddress: 'address456' }] };
+    expect(getUserHandle(user.primaryZID, user.wallets)).toEqual('zid123');
+  });
+
+  it('returns truncated publicAddress from the first wallet when primaryZID is absent', () => {
+    const user = { primaryZID: null, wallets: [{ id: 'wallet-id-1', publicAddress: '123456789' }] };
+    expect(getUserHandle(user.primaryZID, user.wallets)).toEqual('1234...6789');
+  });
+
+  it('returns empty string when both primaryZID and wallets are absent', () => {
+    const user = { primaryZID: null, wallets: [] };
+    expect(getUserHandle(user.primaryZID, user.wallets)).toEqual('');
```

Comment:
a `nit` but i think `getUserHandle` should just take in the `currentUser` & return the `handle` 🤔

Reply:
We should probably only pass in the necessary data rather than the whole `currentUser`